src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -473,5 +473,10 @@ public static int DecodeInPlace(Span<byte> buffer)
         }
 
         #endregion
+
+        public enum ReturnState
```

Comment:
This is a WIP and I do not intend to merge this PR. My intention was to just share the perf tests and data. There is a lot of leftover code here where I was trying things (which doesn't need to be reviewed).